### ISSUE TITLE
[Quick 7] Regression: No Longer Preserving Casted Value After Saving Entity

### ISSUE THREAD

**Issue by homestar9** [coldbox-modules/quick#203]
In 4.1.1, 4.1.2, Quick was updated to preserve casted values after saving.
However, in v5 (as of 5.2.1), Quick no longer preserves the value.

How to replicate
```
// create entity that uses any cast (e.g. BooleanCast@quick)
property name="foo" casts="BooleanCast@quick";

// in your handler, set the value to "0" (a string representation of 0)
getInstance( "myEntity" ).setFoo( "0" );

// dump out the memento
writeDump( serializeJson( myEntity.getMemento() ) ); // foo will be "0"

// persist the entity
myEntity.save();

// dump out the memento
writeDump( serializeJson( myEntity.getMemento() ) ); // foo will still be "0" in Quick 5, but will be false in Quick 4
```

Quick 4 handles the cast properly when persisting, but Quick 5 doesn't seem to recast anymore.

**Quick 4:**
![image](https://user-images.githubusercontent.com/5341142/195949824-39d93d14-14e2-45a6-acd9-a89ceff20bd7.png)

**Quick 5:**
![image](https://user-images.githubusercontent.com/5341142/195949670-595fb910-5a0a-4672-bd12-d3c7ff67fc32.png)

**Comment by homestar9** [comment]
This issue still persists from Quick 5 through current as of today, Quick 7.2.0

**Comment by homestar9** [comment]
Can confirm this was fixed in Quick 9.0.0
